sensor/common/deduper/deduper.go

```diff
@@ -1,43 +1,128 @@
 package deduper
 
 import (
+	"fmt"
 	"reflect"
+	"strings"
 
+	"github.com/pkg/errors"
 	"github.com/stackrox/rox/generated/internalapi/central"
 	"github.com/stackrox/rox/pkg/alert"
 	"github.com/stackrox/rox/pkg/logging"
+	eventPkg "github.com/stackrox/rox/pkg/sensor/event"
 	"github.com/stackrox/rox/pkg/sensor/hash"
 	"github.com/stackrox/rox/sensor/common/managedcentral"
 	"github.com/stackrox/rox/sensor/common/messagestream"
 )
 
 var (
 	log = logging.LoggerForModule()
+
+	deduperTypes = []any{
+		&central.SensorEvent_NetworkPolicy{},
+		&central.SensorEvent_Deployment{},
+		&central.SensorEvent_Pod{},
+		&central.SensorEvent_Namespace{},
+		&central.SensorEvent_Secret{},
+		&central.SensorEvent_Node{},
+		&central.SensorEvent_NodeInventory{},
+		&central.SensorEvent_ServiceAccount{},
+		&central.SensorEvent_Role{},
+		&central.SensorEvent_Binding{},
+		&central.SensorEvent_ProcessIndicator{},
+		&central.SensorEvent_ProviderMetadata{},
+		&central.SensorEvent_OrchestratorMetadata{},
+		&central.SensorEvent_ImageIntegration{},
+		&central.SensorEvent_ComplianceOperatorResult{},
+		&central.SensorEvent_ComplianceOperatorProfile{},
+		&central.SensorEvent_ComplianceOperatorRule{},
+		&central.SensorEvent_ComplianceOperatorScanSettingBinding{},
+		&central.SensorEvent_ComplianceOperatorScan{},
+	}
 )
 
+// Key is an interface to the key by which the messages are deduped.
+type Key interface {
+	GetID() string
+	GetType() reflect.Type
+}
+
 // key is the key by which messages are deduped.
 type key struct {
 	id           string
 	resourceType reflect.Type
 }
 
+func (k key) GetID() string {
+	return k.id
+}
+
+func (k key) GetType() reflect.Type {
+	return k.resourceType
+}
+
+func keyFrom(v string) (Key, error) {
```

Comment:
Could we write some unit tests for this? I'd like to see that the actual examples from deduper states (e.g.: `Binding:d4441ace-ea04-4d5b-8693-97d57f10dace -> 6133788155869413633`) would work.

Reply:
Done in #8078